Date: Thu, 19 Oct 2017 20:11:57 -0700 (PDT)
From: Bili Dong <qobi...@gmail.com>
To: singularity <singu...@lbl.gov>
Message-Id: <799b041c-65cd-4e84-9b3a-46a8ac3fcc83@lbl.gov>
In-Reply-To: <CAAQzouPgh9hmFXM6WSjtpLBcjLSPYYymRKXyfwuCirPTyNO+Fg@mail.gmail.com>
References: <6b519f4e-6afc-4164-a1bf-d8f3eb8fd372@lbl.gov>
 <CAAQzouPgh9hmFXM6WSjtpLBcjLSPYYymRKXyfwuCirPTyNO+Fg@mail.gmail.com>
Subject: Re: [Singularity] Issue with the Miniconda Docker image or using
 Conda in Singularity in general
MIME-Version: 1.0
Content-Type: multipart/mixed; 
	boundary="----=_Part_2189_521633314.1508469118184"

------=_Part_2189_521633314.1508469118184
Content-Type: multipart/alternative; 
	boundary="----=_Part_2190_263320001.1508469118185"

------=_Part_2190_263320001.1508469118185
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit

Thanks Chris. This does look like the same issue.

Best,
Bili

On Thursday, October 19, 2017 at 8:04:33 PM UTC-7, Chris Filo Gorgolewski 
wrote:
>
> This seems to be related: 
> https://github.com/poldracklab/fmriprep/issues/780
>
> Best,
> Chris
>
> On Thu, Oct 19, 2017 at 7:08 PM, Bili Dong <qo...@gmail.com 
> <javascript:>> wrote:
>
>> Hello all,
>>
>> I'm writing to request some help on an issue I encountered that I 
>> couldn't solve by myself.
>>
>> I'm using Singularity 2.4. To reproduce the error, do the following:
>>
>> ```
>> singularity shell docker://continuumio/miniconda3:4.3.27
>> ```
>>
>> then within Singularity, do:
>>
>> ```
>> python -c "import multiprocessing as mp; mp.Queue()"
>> ```
>>
>> which results in the following error:
>>
>> ```
>> Traceback (most recent call last):
>>   File "<string>", line 1, in <module>
>>   File "/opt/conda/lib/python3.6/multiprocessing/context.py", line 102, 
>> in Queue
>>     return Queue(maxsize, ctx=self.get_context())
>>   File "/opt/conda/lib/python3.6/multiprocessing/queues.py", line 42, in 
>> __init__
>>     self._rlock = ctx.Lock()
>>   File "/opt/conda/lib/python3.6/multiprocessing/context.py", line 67, in 
>> Lock
>>     return Lock(ctx=self.get_context())
>>   File "/opt/conda/lib/python3.6/multiprocessing/synchronize.py", line 
>> 163, in __init__
>>     SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)
>>   File "/opt/conda/lib/python3.6/multiprocessing/synchronize.py", line 
>> 60, in __init__
>>     unlink_now)
>> FileNotFoundError: [Errno 2] No such file or directory
>> ```
>>
>> The same code would work inside a Docker container. So I expected it to 
>> work in the corresponding Singularity container too. But it doesn't.
>>
>> Thanks very much for your time!
>>
>> Best,
>> Bili
>>
>> -- 
>> You received this message because you are subscribed to the Google Groups 
>> "singularity" group.
>> To unsubscribe from this group and stop receiving emails from it, send an 
>> email to singu...@lbl.gov <javascript:>.
>>
>
>
------=_Part_2190_263320001.1508469118185
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable

<div dir=3D"ltr">Thanks Chris. This does look like the same issue.<div><br>=
</div><div>Best,</div><div>Bili<br><br>On Thursday, October 19, 2017 at 8:0=
4:33 PM UTC-7, Chris Filo Gorgolewski wrote:<blockquote class=3D"gmail_quot=
e" style=3D"margin: 0;margin-left: 0.8ex;border-left: 1px #ccc solid;paddin=
g-left: 1ex;"><div dir=3D"ltr">This seems to be related:=C2=A0<a href=3D"ht=
tps://github.com/poldracklab/fmriprep/issues/780" target=3D"_blank" rel=3D"=
nofollow" onmousedown=3D"this.href=3D&#39;https://www.google.com/url?q\x3dh=
ttps%3A%2F%2Fgithub.com%2Fpoldracklab%2Ffmriprep%2Fissues%2F780\x26sa\x3dD\=
x26sntz\x3d1\x26usg\x3dAFQjCNHw0gZCuBiDzuYOna29GJ8IbaoyPA&#39;;return true;=
" onclick=3D"this.href=3D&#39;https://www.google.com/url?q\x3dhttps%3A%2F%2=
Fgithub.com%2Fpoldracklab%2Ffmriprep%2Fissues%2F780\x26sa\x3dD\x26sntz\x3d1=
\x26usg\x3dAFQjCNHw0gZCuBiDzuYOna29GJ8IbaoyPA&#39;;return true;">https://gi=
thub.com/<wbr>poldracklab/fmriprep/issues/<wbr>780</a></div><div><br clear=
=3D"all"><div><div><div dir=3D"ltr">Best,<div>Chris</div></div></div></div>
<br><div class=3D"gmail_quote">On Thu, Oct 19, 2017 at 7:08 PM, Bili Dong <=
span dir=3D"ltr">&lt;<a href=3D"javascript:" target=3D"_blank" gdf-obfuscat=
ed-mailto=3D"fsZ7EaQPBgAJ" rel=3D"nofollow" onmousedown=3D"this.href=3D&#39=
;javascript:&#39;;return true;" onclick=3D"this.href=3D&#39;javascript:&#39=
;;return true;">qo...@gmail.com</a>&gt;</span> wrote:<br><blockquote class=
=3D"gmail_quote" style=3D"margin:0 0 0 .8ex;border-left:1px #ccc solid;padd=
ing-left:1ex"><div dir=3D"ltr">Hello all,<div><br></div><div>I&#39;m writin=
g to request some help on an issue I encountered that I couldn&#39;t solve =
by myself.<br><div><br></div><div>I&#39;m using Singularity 2.4. To reprodu=
ce the error, do the following:<div><br></div><div><font face=3D"courier ne=
w, monospace">```</font></div><div><font face=3D"courier new, monospace">si=
ngularity shell=C2=A0docker://continuumio/<wbr>miniconda3:4.3.27</font></di=
v><div><font face=3D"courier new, monospace">```</font></div><div><br></div=
><div>then within Singularity, do:</div><div><br></div><div><font face=3D"c=
ourier new, monospace">```</font></div><div><font face=3D"courier new, mono=
space">python -c &quot;import multiprocessing as mp; mp.Queue()&quot;</font=
></div><div><font face=3D"courier new, monospace">```</font></div><div><br>=
</div><div>which results in the following error:</div><div><br></div><div><=
font face=3D"courier new, monospace">```</font></div><div><font face=3D"cou=
rier new, monospace">Traceback (most recent call last):</font></div><div><f=
ont face=3D"courier new, monospace">=C2=A0 File &quot;&lt;string&gt;&quot;,=
 line 1, in &lt;module&gt;</font></div><div><font face=3D"courier new, mono=
space">=C2=A0 File &quot;/opt/conda/lib/python3.6/<wbr>multiprocessing/cont=
ext.py&quot;, line 102, in Queue</font></div><div><font face=3D"courier new=
, monospace">=C2=A0 =C2=A0 return Queue(maxsize, ctx=3Dself.get_context())<=
/font></div><div><font face=3D"courier new, monospace">=C2=A0 File &quot;/o=
pt/conda/lib/python3.6/<wbr>multiprocessing/queues.py&quot;, line 42, in __=
init__</font></div><div><font face=3D"courier new, monospace">=C2=A0 =C2=A0=
 self._rlock =3D ctx.Lock()</font></div><div><font face=3D"courier new, mon=
ospace">=C2=A0 File &quot;/opt/conda/lib/python3.6/<wbr>multiprocessing/con=
text.py&quot;, line 67, in Lock</font></div><div><font face=3D"courier new,=
 monospace">=C2=A0 =C2=A0 return Lock(ctx=3Dself.get_context())</font></div=
><div><font face=3D"courier new, monospace">=C2=A0 File &quot;/opt/conda/li=
b/python3.6/<wbr>multiprocessing/synchronize.<wbr>py&quot;, line 163, in __=
init__</font></div><div><font face=3D"courier new, monospace">=C2=A0 =C2=A0=
 SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=3Dctx)</font></div><div><font =
face=3D"courier new, monospace">=C2=A0 File &quot;/opt/conda/lib/python3.6/=
<wbr>multiprocessing/synchronize.<wbr>py&quot;, line 60, in __init__</font>=
</div><div><font face=3D"courier new, monospace">=C2=A0 =C2=A0 unlink_now)<=
/font></div><div><font face=3D"courier new, monospace">FileNotFoundError: [=
Errno 2] No such file or directory</font></div><div><font face=3D"courier n=
ew, monospace">```</font></div><div><br></div><div>The same code would work=
 inside a Docker container. So I expected it to work in the corresponding S=
ingularity container too. But it doesn&#39;t.</div><div><br></div><div>Than=
ks very much for your time!</div></div></div><div><br></div><div>Best,</div=
><div>Bili</div></div><span><font color=3D"#888888">

<p></p>

-- <br>
You received this message because you are subscribed to the Google Groups &=
quot;singularity&quot; group.<br>
To unsubscribe from this group and stop receiving emails from it, send an e=
mail to <a href=3D"javascript:" target=3D"_blank" gdf-obfuscated-mailto=3D"=
fsZ7EaQPBgAJ" rel=3D"nofollow" onmousedown=3D"this.href=3D&#39;javascript:&=
#39;;return true;" onclick=3D"this.href=3D&#39;javascript:&#39;;return true=
;">singularity...@lbl.<wbr>gov</a>.<br>
</font></span></blockquote></div><br></div>
</blockquote></div></div>
------=_Part_2190_263320001.1508469118185--

------=_Part_2189_521633314.1508469118184--
